src/EditorFeatures/CSharpTest/TypeInferrer/TypeInferrerTests.cs

```diff
@@ -1609,7 +1609,7 @@ class C
         bool x = await [|M|].ContinueWith(a => { return true; }).ContinueWith(a => { return false; });
     }
 }";
-            await TestAsync(text, "global::System.Threading.Tasks.Task<System.Boolean>");
+            await TestAsync(text, "System.Object", testPosition: false);
```

Comment:
Finding "Task" when someone writes "ContinueWith" is akin to us finding types like ```List<T>``` when the user types "foo.Count".  I think it could be hugely useful.  But it needs to be done properly.  i.e. we should actually be finding types that contain that member and not just saying "i'm *somewhere deep in an await* ergo, my type is the same as the contextual type the await would find".   

To be clear:

```string s = await Foo().ContinueWith(list => list.Count.ToString())```

It is simply wrong for us to say that "Foo()" is a ```Task<string>```.  The only reason this was ok in the test was because the test was contrived.  It happened to be ok in this one case.  But htat's like saying:   "string s = x.y.z.Florb()" and saying that x, y, or z should have the "string" type.  It's just taking a type we see and drawing a very erroneous conclusion about it.

Similarly, if you had "if (a.b)" then "b" should have type 'bool', but 'a' should have nothing inferred about it**.

Up until #643 was "fixed" teh type inference service operated in a very considered and careful manner.  It infers types when there is a genuine connection between things.  i.e. if i have "string s = a.b.c" then "c" def can be inferred to have the "string" type.  similarly "string s = await foo" or "string s = await foo.ConfigureAwait()".  in both these cases 'foo' can be inferred to have type string.  

--

** in the future it would be nifty to say "try to find a type that has a 'b' on it, and pick something suitable".  Until then though, i don't want this bad inference code in the type inferrer.